substratools/function.py

```diff
@@ -175,82 +160,11 @@ def _user_func(args):
     return parser
 
 
-class Algo(GenericAlgo):
-    @abc.abstractmethod
-    def train(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-    @abc.abstractmethod
-    def predict(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-
-class CompositeAlgo(GenericAlgo):
-    @abc.abstractmethod
-    def train(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-    @abc.abstractmethod
-    def predict(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-
-class AggregateAlgo(GenericAlgo):
-    @abc.abstractmethod
-    def aggregate(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-    @abc.abstractmethod
-    def predict(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
-
-
-class MetricAlgo(GenericAlgo):
-    @abc.abstractmethod
-    def score(
-        self,
-        inputs: dict,
-        outputs: dict,
-        task_properties: dict,
-    ) -> None:
-
-        raise NotImplementedError
+def _get_function_from_name(functions_list, function_name):
+    for function in functions_list:
+        if function.__name__ == function_name:
+            return function
+    raise
```

Comment:
raise should be a custom error

Reply:
maybe you can do stgh like 

```py
try:
    eval(function_name) in functions_list
except NameError:
    raise custom exception

Reply:
After some tests, it does not work as the function is not loaded in memory as a variable.